src/Analyser/NodeScopeResolver.php

```diff
@@ -3928,6 +3930,49 @@ private function processClosureNode(
 		?Type $passedToType,
 	): ProcessClosureResult
 	{
+		if ($stmt instanceof Node\Stmt\Expression && $stmt->expr instanceof Expr\CallLike) {
+			if ($stmt->expr instanceof FuncCall && $stmt->expr->name instanceof Name && $this->reflectionProvider->hasFunction($stmt->expr->name, $scope)) {
```

Comment:
We could get rid of these `if` statements if this logic was implemented in `processArgs` (where `processClosureNode` is called) it could be simplified a lot. You wouldn't have to look for the right reflection again, because it's available there as `$calleeReflection`.

Reply:
Sure. I can try that.

Reply:
ea4494680a8aaee57c03530e69375660aea7999e

Yeah it simplified a lot. But the test case with calling static method on a class string doesn't work. And maybe it doesn't need to work 🤷🏽

Reply:
Yeah, I'm not surprised by that. $calleeReflecrion is probably null for that case now.

Reply:
Removed it for now.

Also I forgot to mention, there was no easy way to access the expression in `processArgs`, so had to refactor a bit so it accepts the expression (CallLike) now and extracts the args inside.